stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanSheet.kt

```diff
@@ -79,6 +87,51 @@ class CardScanSheet private constructor(private val stripePublishableKey: String
                 ?: CardScanSheetResult.Failed(
                     UnknownScanException("No data in the result intent")
                 )
+
+        /**
+         * Attach the cardscan fragment to the specified container.
+         * Results will be returned in the callback function.
+         */
+        fun attachCardScanFragment(
+            lifecycleOwner: LifecycleOwner,
+            supportFragmentManager: FragmentManager,
+            @IdRes fragmentContainer: Int,
+            stripePublishableKey: String,
+            onFinished: (cardScanSheetResult: CardScanSheetResult?) -> Unit
+        ) {
+            supportFragmentManager.commit {
+                setReorderingAllowed(true)
+                add<CardScanFragment>(
+                    fragmentContainer,
+                    args = bundleOf(
+                        CARD_SCAN_FRAGMENT_PARAMS_KEY to CardScanSheetParams(stripePublishableKey)
+                    ),
+                    tag = CARD_SCAN_FRAGMENT_TAG
+                )
+            }
+
+            supportFragmentManager
+                .setFragmentResultListener(
+                    CARD_SCAN_FRAGMENT_REQUEST_KEY, lifecycleOwner
+                ) { _, bundle ->
+                    val result: CardScanSheetResult? = bundle.getParcelable(
+                        CARD_SCAN_FRAGMENT_BUNDLE_KEY
+                    )
```

Comment:
Let's make `result` non-nullable. If result is null, default it to a `Failed` with an exception indicating that the bundle was missing the result.